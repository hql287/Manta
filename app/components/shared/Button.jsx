// Libraries
import React from 'react';
import PropTypes from 'prop-types';

// Styles
import styled from 'styled-components';

const ButtonStyle = styled.button`
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  border: none;
  border-radius: 4px;
  padding: 4px 15px;
  font-size: 12px;
  text-decoration: none;
  background: #ffffff;
  border: 1px solid #e0e1e1;
  text-transform: uppercase;
  letter-spacing: 1px;
  // Block Level Button
  ${props => props.block && `width: 100%;`}
  // Color
  ${props => props.primary &&  `
    background: #469fe5;
    color: white;
  `}
  ${props => props.success && `
    background: #6bbb69;
    color: white;
  `}
  ${props => props.danger && `
    background: #EC476E;
    color: white;
  `}
  // Active state
  ${props => props.active && `
    background: #F2F3F4;
    color: #4F555C;
  `}
  // Hover
  &:hover {
    cursor: pointer;
    text-decoration: none;
    // color: white;
  }
`;

const ButtonLinkStyle = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  background: none;
  border: none;
  padding: 0;
  margin: 0;
  ${props => props.primary && `color: #469fe5;`} ${props =>
  props.success && `color: #6bbb69;`} ${props =>
  props.danger && `color: #EC476E;`} &:hover {
    cursor: pointer;
  }
`;

const ButtonsGroupStyle = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-start;
  > button {
    margin: 0!important;
    border-radius: 0;
    &:first-child { border-radius: 4px 0 0 4px; }
    &:last-child { border-radius: 0 4px 4px 0; }
    &:not(:first-child) { border-left: 0; }
  }
`;

function Button(props) {
  return props.link ? (
    <ButtonLinkStyle {...props}>{props.children}</ButtonLinkStyle>
  ) : (
    <ButtonStyle {...props}>{props.children}</ButtonStyle>
  );
}

Button.propTypes = {
  link: PropTypes.bool,
  danger: PropTypes.bool,
  onClick: PropTypes.func,
  primary: PropTypes.bool,
  success: PropTypes.bool,
};

Button.defaultProps = {
  link: false,
  primary: false,
  success: false,
  danger: false,
};

export const ButtonsGroup = props => (
  <ButtonsGroupStyle>{props.children}</ButtonsGroupStyle>
);

export default Button;
