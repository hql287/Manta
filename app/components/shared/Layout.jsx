// Libraries
import React from 'react';

// Styles
import styled from 'styled-components';

const AppWrapperStyle = styled.div`
  display: flex;
  flex-direction: row;
  height: 100%;
`;

const AppMainContentStyle = styled.div`
  overflow: auto;
  width: 100%;
  height: 100%;
  background: #f9fafa;
`;

const PageWrapperStyle = styled.div`
  overflow: auto;
  display: flex;
  flex-direction: column;
  height: 100%;
`;

const PageHeaderStyle = styled.div`
  position: fixed;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 50px;
  padding: 10px 120px 10px 40px;
  background: #fff;
  border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  z-index: 1;
`;

const PageHeaderTitleStyle = styled.p`
  font-size: 18px;
  font-weight: 600;
  margin: 0;
  letter-spacing: -0.5px;
`;

const PageHeaderActionsStyle = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: flex-end;
  button {  margin-left: 10px; }
  i { margin-right: 10px; }
`;

const PageContentStyle = styled.div`
  // flex: 1;
  // overflow: hidden;
  margin: 90px 40px 40px 40px;
  ${props =>
    !props.bare &&
    `
    border: 1px solid rgba(0,0,0,.1);
    border-radius: 4px;
    background: #FFF;
  `};
`;

const PageFooterStyle = styled.div`
  position: fixed;
  width: 100%;
  bottom: 0;
  background: #f9fafa;
  border-top: 1px solid rgba(0, 0, 0, 0.1);
`;

// Components
const AppWrapper = props => <AppWrapperStyle>{props.children}</AppWrapperStyle>;

const AppMainContent = props => (
  <AppMainContentStyle>{props.children}</AppMainContentStyle>
);

const PageWrapper = props => (
  <PageWrapperStyle>{props.children}</PageWrapperStyle>
);

const PageHeader = props => <PageHeaderStyle>{props.children}</PageHeaderStyle>;

const PageHeaderTitle = props => (
  <PageHeaderTitleStyle>{props.children}</PageHeaderTitleStyle>
);

const PageHeaderActions = props => (
  <PageHeaderActionsStyle>{props.children}</PageHeaderActionsStyle>
);

const PageContent = props => (
  <PageContentStyle bare={props.bare}>{props.children}</PageContentStyle>
);

const PageFooter = props => <PageFooterStyle>{props.children}</PageFooterStyle>;

export {
  AppWrapper,
  AppMainContent,
  PageWrapper,
  PageHeader,
  PageHeaderTitle,
  PageHeaderActions,
  PageContent,
  PageFooter,
};
