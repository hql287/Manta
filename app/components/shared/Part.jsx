// Libraries
import React from 'react';
import PropTypes from 'prop-types';

// Styles
import styled from 'styled-components';

const PartStyle = styled.div`
  padding: 20px 20px 10px 20px;
  background: #f9fafa;
  border-radius: 4px;
  margin-bottom: 20px;
  border: 1px solid #f2f3f4;
`;

const RowStyle = styled.div`
  display: flex;
  margin: 0 -15px;
`;

const FieldStyle = styled.div`
  flex: 1;
  margin: 0 15px 20px 15px;
`;

const HeaderStyle = styled.h2``;


// Components
export const Part = props => <PartStyle>{props.children}</PartStyle>;
export const Header = props => <HeaderStyle>{props.children}</HeaderStyle>;
export const Field = props => <FieldStyle>{props.children}</FieldStyle>;
export const Row = props => <RowStyle>{props.children}</RowStyle>;
