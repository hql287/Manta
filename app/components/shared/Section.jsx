// Libraries
import React from 'react';
import PropTypes from 'prop-types';

// Styles
import styled from 'styled-components';

const SectionStyle = styled.div`
  position: relative;
  display: block;
  margin-bottom: 30px;
  padding: 0 40px;
`;

const HeaderStyle = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  a {
    font-size: 14px;
    line-height: 12px;
    // margin-bottom: 20px;
    margin-left: 10px;
  }
`;

// Components
export const Section = props => <SectionStyle>{props.children}</SectionStyle>;
export const Header = props => <HeaderStyle>{props.children}</HeaderStyle>;
