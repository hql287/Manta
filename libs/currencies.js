import currencies from './currencies.json';
import cryptocurrencies from 'cryptocurrencies';

const Crypto = () => {
    let cryptocurrencie = {};

    const keys = Object.keys(cryptocurrencies)

    for (let index = 0; index < keys.length; index++) {
        
        cryptocurrencie[keys[index]] = {
            "symbol": keys[index],
            "name":  cryptocurrencies[keys[index]],
            "symbol_native": keys[index],
            "decimal_digits": 2,
            "rounding": 0,
            "code": keys[index],
            "name_plural": ""
        }
    }

    return cryptocurrencie;
}

const All = () => (Object.assign({}, currencies, Crypto()))

const List = () => ({ currencies, cryptocurrencies: Crypto()});

export {
    All,
    List
}