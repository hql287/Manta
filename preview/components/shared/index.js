// Libs
import React from 'react';

import styled from 'styled-components';

const Section = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 30px;
  justify-content: flex-end;
  align-items: flex-start;
  &:last-child {
    margin-bottom: 0;
  }
`;

const Label = styled.label`
  font-size: 12px;
  text-transform: uppercase;
  letter-spacing: 1px;
  font-weight: 600;
  color: #4f555c;
`;

const Range = styled.input`
  width: 100%;
  -webkit-app-region: no-drag;
`;

export { Section, Label, Range };
