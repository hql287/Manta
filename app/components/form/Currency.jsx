// Libs
import React, {Component} from 'react';
import PropTypes from 'prop-types';
const appConfig = require('electron').remote.require('electron-settings');
import currencies from '../../../libs/currencies.json';
import { keys, sortBy } from 'lodash';

// Custom Components
import {Section} from '../shared/Section';

// Animation
import _withFadeInAnimation from '../shared/hoc/_withFadeInAnimation';

// Component
export class Currency extends Component {
  constructor(props) {
    super(props);
    this.updateCurrency = this.updateCurrency.bind(this);
  }

  componentDidMount() {
    const currencyCode = appConfig.get('appSettings.currency');
    this.props.updateFieldData('currency', {
      selectedCurrency: currencies[currencyCode],
    });
  }

  shouldComponentUpdate(nextProps) {
    return this.props.currency !== nextProps.currency;
  }

  updateCurrency(event) {
    const value = event.target.value;
    const currencyCode = value === '' ? null : value;
    this.props.updateFieldData('currency', {
      selectedCurrency: currencies[currencyCode]
    });
  }

  render() {
    const currenciesKeys = keys(currencies);
    const currenciesKeysAndValues = currenciesKeys.
      map(key => [key, currencies[key]['name'], currencies[key]['code']]);
    const currenciesSorted = sortBy(currenciesKeysAndValues, [array => array[1]]);
    const currenciesOptions = currenciesSorted.map((obj) => {
      const [key, name, code] = obj;

      let optionKey = code;
      let optionValue = code;
      let optionLabel = name;

      return (
        <option value={optionValue} key={optionKey}>
          {optionLabel}
        </option>
      );
    });
    const {currency} = this.props;
    const currencyCode = currency.selectedCurrency
      ? currency.selectedCurrency.code
      : appConfig.get('appSettings.currency');
    return (
      <Section>
        <label className="itemLabel">Currency</label>
        <select value={currencyCode} onChange={this.updateCurrency}>
          {currenciesOptions}
        </select>
      </Section>
    );
  }
}

Currency.propTypes = {
  currency: PropTypes.object.isRequired,
  updateFieldData: PropTypes.func.isRequired,
};

// Export
export default _withFadeInAnimation(Currency);
