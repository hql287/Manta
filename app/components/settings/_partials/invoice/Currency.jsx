import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { keys, sortBy } from 'lodash';

import currencies from '../../../../../libs/currencies.json';
import { Row, Field, Part } from '../../../shared/Part';

function sortCurrencies() {
  const currenciesKeys = keys(currencies);
  const currenciesKeysAndValues = currenciesKeys.map(key => [
    key,
    currencies[key].name,
    currencies[key].code,
  ]);
  const currenciesSorted = sortBy(currenciesKeysAndValues, [array => array[1]]);
  return currenciesSorted.map(obj => {
    const [key, name, code] = obj;
    const optionKey = code;
    const optionValue = code;
    const optionLabel = name;
    return (
      <option value={optionValue} key={optionKey}>
        {optionLabel}
      </option>
    );
  });
}

function Currency({ t, currency, handleCurrencyChange }) {
  const { code, fraction, placement, separator } = currency;
  return [
    <label key="label" className="itemLabel">
      {t('form:fields:currency')}
    </label>,
    <Part key="part">
      <Row>
        <Field>
          <label className="itemLabel">{t('form:fields:currency')}</label>
          <select
            name="code"
            value={code}
            onChange={handleCurrencyChange}
          >
            {sortCurrencies()}
          </select>
        </Field>
        <Field>
          <label className="itemLabel">{t('settings:fields:currency:separator')}</label>
          <select
            name="separator"
            value={separator}
            onChange={handleCurrencyChange}
          >
            <option value="commaDot">1,999.000 (Comma & Dot )</option>
            <option value="dotComma">1.999,000 (Dot & Comma)</option>
            <option value="spaceDot">1 999.000 (Space & Dot)</option>
          </select>
        </Field>
      </Row>
      <Row>
        <Field>
          <label className="itemLabel">{t('settings:fields:currency:placement')}</label>
          <select
            name="placement"
            value={placement}
            onChange={handleCurrencyChange}
          >
            <option value="before">{t('settings:fields:currency:beforeAmount')}</option>
            <option value="after">{t('settings:fields:currency:afterAmount')}</option>
          </select>
        </Field>
        <Field>
          <label className="itemLabel">{t('settings:fields:currency:fraction')}</label>
          <input
            className="form-control"
            name="fraction"
            type="number"
            value={fraction}
            onChange={handleCurrencyChange}
          />
        </Field>
      </Row>
    </Part>,
  ];
}

Currency.propTypes = {
  currency: PropTypes.object.isRequired,
  handleCurrencyChange: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
};

export default Currency;
