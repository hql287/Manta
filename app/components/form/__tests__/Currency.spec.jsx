// Libs
import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { All as currencies } from '../../../../libs/currencies.js';
const appConfig = require('electron').remote.require('electron-settings');

// Component
import { Currency } from '../Currency';
import { Section } from '../../shared/Section';

// Mocks
const currency = currencies.USD;
const savedSettings = 'USD';
const updateFieldData = jest.fn();
const updateSavedSettings = jest.fn();

describe('Renders correctly to the DOM', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <Currency
        currency={currency}
        savedSettings={savedSettings}
        updateFieldData={updateFieldData}
        updateSavedSettings={updateSavedSettings}
      />
    );
  });

  // TODO
  it('will not rerender if threre is no change in Props', () => {});

  it('renders necessary element', () => {
    // Section
    expect(wrapper.find(Section)).toHaveLength(1);
    expect(wrapper.find(Section)).not.toHaveLength(2);
    // label
    expect(wrapper.find('label')).toHaveLength(1);
    expect(wrapper.find('label')).not.toHaveLength(2);
    // select
    expect(wrapper.find('select')).toHaveLength(1);
    expect(wrapper.find('select')).not.toHaveLength(2);
  });

  it('handle select change correctly', () => {
    const selectEl = wrapper.find('select');
    selectEl.simulate('change', { target: { value: 'VND' } });
    expect(updateFieldData).toHaveBeenCalled();
    expect(updateFieldData).toHaveBeenCalledWith('currency', currencies.VND);
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <Currency
          currency={currency}
          savedSettings={savedSettings}
          updateFieldData={updateFieldData}
          updateSavedSettings={updateSavedSettings}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
