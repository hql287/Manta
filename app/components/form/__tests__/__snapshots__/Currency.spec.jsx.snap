// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly to the DOM matches snapshot 1`] = `
<div
  className="Section__SectionStyle-dgSKSv cwfAAu"
>
  <div
    className="Section__HeaderStyle-chmDx gHBKbw"
  >
    <label
      className="itemLabel"
    />
  </div>
  <div
    className="Part__PartStyle-kAWmVX iwwkMM"
  >
    <div
      className="Part__RowStyle-nLRvo cGeXDT"
    >
      <div
        className="Part__FieldStyle-dkcwja bIqrsv"
      >
        <label
          className="itemLabel"
        />
        <select
          name="code"
          onChange={[Function]}
          value="USD"
        >
          <option
            value="AFN"
          >
            Afghan Afghani
          </option>
          <option
            value="ALL"
          >
            Albanian Lek
          </option>
          <option
            value="DZD"
          >
            Algerian Dinar
          </option>
          <option
            value="ARS"
          >
            Argentine Peso
          </option>
          <option
            value="AMD"
          >
            Armenian Dram
          </option>
          <option
            value="AUD"
          >
            Australian Dollar
          </option>
          <option
            value="AZN"
          >
            Azerbaijani Manat
          </option>
          <option
            value="BHD"
          >
            Bahraini Dinar
          </option>
          <option
            value="BDT"
          >
            Bangladeshi Taka
          </option>
          <option
            value="BYR"
          >
            Belarusian Ruble
          </option>
          <option
            value="BZD"
          >
            Belize Dollar
          </option>
          <option
            value="BOB"
          >
            Bolivian Boliviano
          </option>
          <option
            value="BAM"
          >
            Bosnia-Herzegovina Convertible Mark
          </option>
          <option
            value="BWP"
          >
            Botswanan Pula
          </option>
          <option
            value="BRL"
          >
            Brazilian Real
          </option>
          <option
            value="GBP"
          >
            British Pound Sterling
          </option>
          <option
            value="BND"
          >
            Brunei Dollar
          </option>
          <option
            value="BGN"
          >
            Bulgarian Lev
          </option>
          <option
            value="BIF"
          >
            Burundian Franc
          </option>
          <option
            value="XOF"
          >
            CFA Franc BCEAO
          </option>
          <option
            value="XAF"
          >
            CFA Franc BEAC
          </option>
          <option
            value="KHR"
          >
            Cambodian Riel
          </option>
          <option
            value="CAD"
          >
            Canadian Dollar
          </option>
          <option
            value="CVE"
          >
            Cape Verdean Escudo
          </option>
          <option
            value="CLP"
          >
            Chilean Peso
          </option>
          <option
            value="CNY"
          >
            Chinese Yuan
          </option>
          <option
            value="COP"
          >
            Colombian Peso
          </option>
          <option
            value="KMF"
          >
            Comorian Franc
          </option>
          <option
            value="CDF"
          >
            Congolese Franc
          </option>
          <option
            value="CRC"
          >
            Costa Rican Colón
          </option>
          <option
            value="HRK"
          >
            Croatian Kuna
          </option>
          <option
            value="CZK"
          >
            Czech Republic Koruna
          </option>
          <option
            value="DKK"
          >
            Danish Krone
          </option>
          <option
            value="DJF"
          >
            Djiboutian Franc
          </option>
          <option
            value="DOP"
          >
            Dominican Peso
          </option>
          <option
            value="EGP"
          >
            Egyptian Pound
          </option>
          <option
            value="ERN"
          >
            Eritrean Nakfa
          </option>
          <option
            value="EEK"
          >
            Estonian Kroon
          </option>
          <option
            value="ETB"
          >
            Ethiopian Birr
          </option>
          <option
            value="EUR"
          >
            Euro
          </option>
          <option
            value="GEL"
          >
            Georgian Lari
          </option>
          <option
            value="GHS"
          >
            Ghanaian Cedi
          </option>
          <option
            value="GTQ"
          >
            Guatemalan Quetzal
          </option>
          <option
            value="GNF"
          >
            Guinean Franc
          </option>
          <option
            value="HNL"
          >
            Honduran Lempira
          </option>
          <option
            value="HKD"
          >
            Hong Kong Dollar
          </option>
          <option
            value="HUF"
          >
            Hungarian Forint
          </option>
          <option
            value="ISK"
          >
            Icelandic Króna
          </option>
          <option
            value="INR"
          >
            Indian Rupee
          </option>
          <option
            value="IDR"
          >
            Indonesian Rupiah
          </option>
          <option
            value="IRR"
          >
            Iranian Rial
          </option>
          <option
            value="IQD"
          >
            Iraqi Dinar
          </option>
          <option
            value="ILS"
          >
            Israeli New Sheqel
          </option>
          <option
            value="JMD"
          >
            Jamaican Dollar
          </option>
          <option
            value="JPY"
          >
            Japanese Yen
          </option>
          <option
            value="JOD"
          >
            Jordanian Dinar
          </option>
          <option
            value="KZT"
          >
            Kazakhstani Tenge
          </option>
          <option
            value="KES"
          >
            Kenyan Shilling
          </option>
          <option
            value="KWD"
          >
            Kuwaiti Dinar
          </option>
          <option
            value="LVL"
          >
            Latvian Lats
          </option>
          <option
            value="LBP"
          >
            Lebanese Pound
          </option>
          <option
            value="LYD"
          >
            Libyan Dinar
          </option>
          <option
            value="LTL"
          >
            Lithuanian Litas
          </option>
          <option
            value="MOP"
          >
            Macanese Pataca
          </option>
          <option
            value="MKD"
          >
            Macedonian Denar
          </option>
          <option
            value="MGA"
          >
            Malagasy Ariary
          </option>
          <option
            value="MYR"
          >
            Malaysian Ringgit
          </option>
          <option
            value="MUR"
          >
            Mauritian Rupee
          </option>
          <option
            value="MXN"
          >
            Mexican Peso
          </option>
          <option
            value="MDL"
          >
            Moldovan Leu
          </option>
          <option
            value="MAD"
          >
            Moroccan Dirham
          </option>
          <option
            value="MZN"
          >
            Mozambican Metical
          </option>
          <option
            value="MMK"
          >
            Myanma Kyat
          </option>
          <option
            value="NAD"
          >
            Namibian Dollar
          </option>
          <option
            value="NPR"
          >
            Nepalese Rupee
          </option>
          <option
            value="TWD"
          >
            New Taiwan Dollar
          </option>
          <option
            value="NZD"
          >
            New Zealand Dollar
          </option>
          <option
            value="NIO"
          >
            Nicaraguan Córdoba
          </option>
          <option
            value="NGN"
          >
            Nigerian Naira
          </option>
          <option
            value="NOK"
          >
            Norwegian Krone
          </option>
          <option
            value="OMR"
          >
            Omani Rial
          </option>
          <option
            value="PKR"
          >
            Pakistani Rupee
          </option>
          <option
            value="PAB"
          >
            Panamanian Balboa
          </option>
          <option
            value="PYG"
          >
            Paraguayan Guarani
          </option>
          <option
            value="PEN"
          >
            Peruvian Nuevo Sol
          </option>
          <option
            value="PHP"
          >
            Philippine Peso
          </option>
          <option
            value="PLN"
          >
            Polish Zloty
          </option>
          <option
            value="QAR"
          >
            Qatari Rial
          </option>
          <option
            value="RON"
          >
            Romanian Leu
          </option>
          <option
            value="RUB"
          >
            Russian Ruble
          </option>
          <option
            value="RWF"
          >
            Rwandan Franc
          </option>
          <option
            value="SAR"
          >
            Saudi Riyal
          </option>
          <option
            value="RSD"
          >
            Serbian Dinar
          </option>
          <option
            value="SGD"
          >
            Singapore Dollar
          </option>
          <option
            value="SOS"
          >
            Somali Shilling
          </option>
          <option
            value="ZAR"
          >
            South African Rand
          </option>
          <option
            value="KRW"
          >
            South Korean Won
          </option>
          <option
            value="LKR"
          >
            Sri Lankan Rupee
          </option>
          <option
            value="SDG"
          >
            Sudanese Pound
          </option>
          <option
            value="SEK"
          >
            Swedish Krona
          </option>
          <option
            value="CHF"
          >
            Swiss Franc
          </option>
          <option
            value="SYP"
          >
            Syrian Pound
          </option>
          <option
            value="TZS"
          >
            Tanzanian Shilling
          </option>
          <option
            value="THB"
          >
            Thai Baht
          </option>
          <option
            value="TOP"
          >
            Tongan Paʻanga
          </option>
          <option
            value="TTD"
          >
            Trinidad and Tobago Dollar
          </option>
          <option
            value="TND"
          >
            Tunisian Dinar
          </option>
          <option
            value="TRY"
          >
            Turkish Lira
          </option>
          <option
            value="USD"
          >
            US Dollar
          </option>
          <option
            value="UGX"
          >
            Ugandan Shilling
          </option>
          <option
            value="UAH"
          >
            Ukrainian Hryvnia
          </option>
          <option
            value="AED"
          >
            United Arab Emirates Dirham
          </option>
          <option
            value="UYU"
          >
            Uruguayan Peso
          </option>
          <option
            value="UZS"
          >
            Uzbekistan Som
          </option>
          <option
            value="VEF"
          >
            Venezuelan Bolívar
          </option>
          <option
            value="VND"
          >
            Vietnamese Dong
          </option>
          <option
            value="YER"
          >
            Yemeni Rial
          </option>
          <option
            value="ZMW"
          >
            Zambian Kwacha
          </option>
        </select>
      </div>
      <div
        className="Part__FieldStyle-dkcwja bIqrsv"
      >
        <label
          className="itemLabel"
        />
        <select
          name="separator"
          onChange={[Function]}
          value="commaDot"
        >
          <option
            value="commaDot"
          >
            1,999.000 (
            )
          </option>
          <option
            value="dotComma"
          >
            1.999,000 (
            )
          </option>
          <option
            value="spaceDot"
          >
            1 999.000 (
            )
          </option>
        </select>
      </div>
    </div>
    <div
      className="Part__RowStyle-nLRvo cGeXDT"
    >
      <div
        className="Part__FieldStyle-dkcwja bIqrsv"
      >
        <label
          className="itemLabel"
        />
        <select
          name="placement"
          onChange={[Function]}
          value="before"
        >
          <option
            value="before"
          />
          <option
            value="after"
          />
        </select>
      </div>
      <div
        className="Part__FieldStyle-dkcwja bIqrsv"
      >
        <label
          className="itemLabel"
        />
        <input
          className="form-control"
          name="fraction"
          onChange={[Function]}
          type="number"
          value={2}
        />
      </div>
    </div>
  </div>
</div>
`;
